File: sync/src/main/scala/com/fan/mysql/event/impl/RowData.java
package com.dingcloud.dts.binlog.event.impl;

import java.util.BitSet;

import com.dingcloud.dts.binlog.column.EventColumn;

public class RowData {

	private EventColumn[] beforeColumns;
	private EventColumn[] afterColumns;
	private BitSet beforeBit;
	private BitSet afterBit;

	public EventColumn[] getBeforeColumns() {
		return beforeColumns;
	}

	public void setBeforeColumns(EventColumn[] beforeColumns) {
		this.beforeColumns = beforeColumns;
	}

	public EventColumn[] getAfterColumns() {
		return afterColumns;
	}

	public void setAfterColumns(EventColumn[] afterColumns) {
		this.afterColumns = afterColumns;
	}

	public BitSet getBeforeBit() {
		return beforeBit;
	}

	public void setBeforeBit(BitSet beforeBit) {
		this.beforeBit = beforeBit;
	}

	public BitSet getAfterBit() {
		return afterBit;
	}

	public void setAfterBit(BitSet afterBit) {
		this.afterBit = afterBit;
	}

}

File: async/src/main/scala/com/fan/mysql/async/binlog/event/impl/RowData.scala
package com.fan.mysql.async.binlog.event.impl

import java.util

import com.fan.mysql.async.binlog.event.EventColumn
import org.apache.commons.lang3.builder.{ToStringBuilder, ToStringStyle}

class RowData {
  private var beforeColumns: Array[EventColumn] = _
  private var afterColumns: Array[EventColumn]  = _
  private var beforeBit: util.BitSet            = _
  private var afterBit: util.BitSet             = _

  def getBeforeColumns: Array[EventColumn] = beforeColumns

  def setBeforeColumns(beforeColumns: Array[EventColumn]): Unit = {
    this.beforeColumns = beforeColumns
  }

  def getAfterColumns: Array[EventColumn] = afterColumns

  def setAfterColumns(afterColumns: Array[EventColumn]): Unit = {
    this.afterColumns = afterColumns
  }

  def getBeforeBit: util.BitSet = beforeBit

  def setBeforeBit(beforeBit: util.BitSet): Unit = {
    this.beforeBit = beforeBit
  }

  def getAfterBit: util.BitSet = afterBit

  def setAfterBit(afterBit: util.BitSet): Unit = {
    this.afterBit = afterBit
  }

  override def toString: String =
    ToStringBuilder.reflectionToString(this, ToStringStyle.DEFAULT_STYLE)
}
